File: scripting/engine/Playground/ecs/components/RigidbodyComponent.swift
@component
public struct RigidBodyComponent {
    internal var handle: UInt64 = UInt64.max
    public var mass: Float
    public var damping: Float
    internal var isDirty: Bool = true

    public init(mass: Float, damping: Float) {
        self.mass = mass
        self.damping = damping
    }
}

File: engine/managed/PlaygroundAssembly/ECS/Physics/RigidBodyComponent.cs
using PlaygroundAssembly.ECS.Core;

namespace PlaygroundAssembly.ECS.Physics;

[EcsComponent]
public struct RigidBodyComponent(float mass, float damping)
{
    internal ulong Handle = ulong.MaxValue;
    internal bool IsDirty = true;

    public float Mass = mass;
    public float Damping = damping;
}
